src/contexts/NewDelivery.tsx

```diff
@@ -0,0 +1,189 @@
+import { createContext, ReactNode, useEffect, useState } from 'react'
+import { SubmitHandler } from 'react-hook-form'
+import useNewDeliverySteps from '@/hooks/useNewDeliverySteps'
+import { NewDeliveryContextValues } from '@/interfaces/newDeliveryContextValues'
+import { useTranslation } from 'react-i18next'
+
+interface NewDeliveryProviderProps {
+  children: ReactNode
+}
+
+export const NewDeliveryContext = createContext<NewDeliveryContextValues>({
+  currentStep: 1,
+  formsData: {},
+  openDialog: false,
+  onCloseDialog: () => {},
+  handleBack: () => {},
+  handleClickOpen: () => {},
+  handleSubmit: () => {},
+  updateStep4Item: () => {},
+  deleteStep4Item: () => {},
+  alertMessage: '',
+  isCompletedMove: false,
+  isExceedQuantity: false
+})
+
+interface MoveGood {
+  type: string // palettes | packages | pieces
+  quantity: number
+  zone: string
+}
+
+export default function NewDeliveryProvider({ children }: NewDeliveryProviderProps) {
+  const { t: translate } = useTranslation()
+  const [currentStep, setCurrentStep] = useState(1)
+  const [formsData, setFormsData] = useState<any>({})
+  const [openDialog, setOpenDialog] = useState(false)
+  const steps = useNewDeliverySteps()
+  const [step3Items, setStep3Items] = useState({ pallets: 0, packages: 0, pieces: 0 })
+  const [step4Items, setStep4Items] = useState<MoveGood[]>([{ type: '', quantity: 0, zone: '' }])
+  const [alertMessage, setAlertMessage] = useState('')
+  const [isCompletedMove, setIsCompletedMove] = useState(false)
+  const [isExceedQuantity, setIsExceedQuantity] = useState(false)
+
+  useEffect(() => {
+    // Delete if condition when the newDeliveryProvider is not in main.tsx
+    // Initial set step3Items
+    if (formsData.goods) {
+      const newStep3Items = { ...step3Items }
+      formsData.goods.map((good: any) => {
+        switch (good.goodTypeStep3) {
+          case 'Палети':
+            newStep3Items.pallets = good.goodQuantityStep3
+            return setStep3Items(newStep3Items)
+          case 'Пакети':
+            newStep3Items.packages = good.goodQuantityStep3
+            return setStep3Items(newStep3Items)
+          case 'Бройки':
+            newStep3Items.pieces = good.goodQuantityStep3
+            return setStep3Items(newStep3Items)
+        }
+      })
+    }
+  }, [formsData])
+
+  useEffect(() => {
+    const currentItems = { pallets: 0, packages: 0, pieces: 0 }
+
+    step4Items.map((item) => {
+      if (item.type === translate('newDelivery.goodType.pallets')) {
```

Comment:
keep the type either as an enum or english, dont have it as a translation. Same for in step 3

Reply:
Fixed